modules/xfem/src/base/XFEM.C

```diff
@@ -59,7 +59,7 @@ XFEM::getCrackTipOrigin(std::map<unsigned int, const Elem* > & elem_id_crack_tip
   std::map<unsigned int, const Elem*> elem_id_map;
 
   int m = -1;
-  for (mit1 = _elem_crack_origin_direction_map.begin(); mit1 != _elem_crack_origin_direction_map.end(); mit1++)
+  for (; mit1 != _elem_crack_origin_direction_map.end(); ++mit1)
```

Comment:
In this case, the iterator is being re-used outside this loop, which is why I left it the way it was.  It doesn't have to be re-used. We could create a new one for each iteration in the loop just below this one. That seemed like a bad idea, but I'm open to changing it.